nuxeo-features/nuxeo-platform-audit/nuxeo-platform-audit-core/src/main/java/org/nuxeo/ecm/platform/audit/io/ExtendedInfoSerializer.java

```diff
@@ -57,22 +57,30 @@ public void serialize(ExtendedInfo info, JsonGenerator jg,
             Serializable value = ((ExtendedInfoImpl.BlobInfo) info).getBlobValue();
             jg.writeObject(Base64.encodeBase64(SerializationUtils.serialize(value)));
         } else if (info instanceof ExtendedInfoImpl.StringInfo) {
-            String stringValue = ((ExtendedInfoImpl.StringInfo) info).getStringValue().trim();
-            if ((stringValue.startsWith("{") && stringValue.endsWith("}"))
-                    || (stringValue.startsWith("[") && stringValue.endsWith("]"))) {
-                try {
-                    mapper.readTree(stringValue);
-                    jg.writeRawValue(stringValue);
-                } catch (IOException e) {
-                    // If the value represents an invalid JSON, send a null value to ES to prevent potential
-                    // mapping exceptions
-                    jg.writeObject(null);
-                }
+            writeString(jg, mapper, ((ExtendedInfoImpl.StringInfo) info).getStringValue().trim());
+        } else {
+            Serializable value = info.getSerializableValue();
+            if (value instanceof String) {
+                writeString(jg, mapper, (String) value);
             } else {
-                jg.writeString(stringValue);
+                jg.writeObject(value);
+            }
+        }
+    }
+
+    private void writeString(JsonGenerator jg, ObjectMapper mapper, String stringValue) throws IOException {
+        if ((stringValue.startsWith("{") && stringValue.endsWith("}"))
+                || (stringValue.startsWith("[") && stringValue.endsWith("]"))) {
+            try {
+                mapper.readTree(stringValue);
+                jg.writeRawValue(stringValue);
+            } catch (IOException e) {
+                // If the value represents an invalid JSON, send a null value to ES to prevent potential
+                // mapping exceptions
+                jg.writeObject(null);
```

Comment:
So what happens if I write `{foo{{{{}` as a string in extended info, as a comment for instance. Is it going to be interpreted as JSON, throw IOException and end up being written as null?

Reply:
Exactly, I didn't change logic here and I think it could be risky to change it now whereas we have this code since two LTS. I'm in favor to rework the whole logic around extended info in audit, there's strange stuff like this one, but ony for next version. Wdyt ?